src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Font.kt

```diff
@@ -0,0 +1,397 @@
+package com.chattriggers.ctjs.minecraft.libs.renderer
+
+import com.chattriggers.ctjs.minecraft.wrappers.Client
+import com.chattriggers.ctjs.utils.Config
+import net.minecraft.client.renderer.GlStateManager
+import net.minecraft.util.MathHelper
+import org.lwjgl.BufferUtils
+import org.lwjgl.opengl.GL11
+import org.lwjgl.util.vector.Vector2f
+import java.awt.Color
+import java.awt.Font
+import java.awt.Graphics2D
+import java.awt.RenderingHints
+import java.awt.image.BufferedImage
+import java.io.FileInputStream
+import java.nio.Buffer
+
+class Font(path: String, size: Float) {
+    private val font: Font
+
+    private lateinit var regularData: Array<CharacterData?>
+    private lateinit var boldData: Array<CharacterData?>
+    private lateinit var italicsData: Array<CharacterData?>
+    private var generatedCharData = false
+
+    private val colorCodes = IntArray(32)
+
+    init {
+        val inputStream = FileInputStream("${Config.modulesFolder}/$path")
+        this.font = Font.createFont(0, inputStream).deriveFont(Font.PLAIN, size)
+
+        Client.scheduleTask {
+            regularData = setup(arrayOfNulls(256), Font.PLAIN)
```

Comment:
This seems a bit weird, couldn't the array just be constructed inside `setup`? Should also allow the elements to be non-nullable types.